src/sync/app.cpp

```diff
@@ -811,13 +811,58 @@ void App::refresh_access_token(std::shared_ptr<SyncUser> sync_user,
     }, handler);
 }
 
+void App::call_function(const std::string& name,
+                        const bson::BsonArray& args_bson,
+                        const util::Optional<std::string>& service_name,
+                        std::function<void (util::Optional<AppError>,
+                                            util::Optional<bson::Bson>)> completion_block) const
+{
+    auto handler = [completion_block](const Response& response) {
+        if (auto error = check_for_errors(response)) {
+            return completion_block(error, util::none);
+        }
+
+        completion_block(util::none, util::Optional<bson::Bson>(bson::parse(response.body)));
+    };
+
+    std::string route = util::format("%1/app/%2/functions/call", m_base_route, m_config.app_id);
+
+    bson::BsonDocument args {
+        { "arguments", args_bson },
+        { "name", name }
+    };
+
+    if (service_name) {
+        args["service"] = *service_name;
+    }
+
+    std::stringstream s;
+    s << bson::Bson(args);
+
+    Request request {
+        .method = HttpMethod::post,
+        .url = route,
+        .body = s.str()
+    };
+
+    do_authenticated_request(request,
+                             SyncManager::shared().get_current_user(),
```

Comment:
We need to be able to call functions for a specific user and not only a current sync user, so maybe just pass the user as an argument.

Reply:
I've added the ability to pass a user in call_function